src/oaklib/implementations/bioportal/agroportal_implementation.py

```diff
@@ -1,23 +1,18 @@
 from dataclasses import dataclass
 
-from oaklib.implementations import BioportalImplementation
-from oaklib.interfaces.mapping_provider_interface import MappingProviderInterface
-from oaklib.interfaces.search_interface import SearchInterface
-from oaklib.interfaces.text_annotator_interface import TextAnnotatorInterface
+from ontoportal_client import AgroPortalClient
+
+from oaklib.implementations.ontoportal.ontoportal_implementation_base import (
+    OntoPortalImplementationBase,
+)
 
 
 @dataclass
-class AgroportalImplementation(
-    BioportalImplementation, TextAnnotatorInterface, SearchInterface, MappingProviderInterface
-):
+class AgroportalImplementation(OntoPortalImplementationBase):
     """
     Implementation over agroportal endpoint
 
     """
 
-    @property
-    def _base_url(self) -> str:
-        return "http://data.agroportal.lirmm.fr/"
-
-    def load_bioportal_api_key(self, path: str = None) -> None:
-        self.bioportal_api_key = "1de0a270-29c5-4dda-b043-7c3580628cd5"
+    OntoPortalClientClass = AgroPortalClient
```

Comment:
would suggest following pep8 naming conventions and call this `ontoportal_client_class` instead

Reply:
JavaScript habits are hard to shake 😁 Will fix.

Reply:
there are code quality checks to ensure we don't do stuff like this by accident, some of them are just taking longer to put in place in earnest in this project because it already has so much technical debt